test/NuGet.Core.Tests/NuGet.Packaging.Test/ContentModelTests/PatternExpressionMatchTests.cs

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using NuGet.ContentModel;
+using NuGet.ContentModel.Infrastructure;
+using Xunit;
+
+namespace NuGet.Client.Test
+{
+    public class PatternExpressionMatchTests
+    {
+
+        
+        [Fact]
+        public void LiteralSegment_OnlyShouldMatch()
+        {
+            // match literal segments given a pattern
+            Dictionary<string, ContentPropertyDefinition> propertyDefinitions =
+                new Dictionary<string, ContentPropertyDefinition>() {
+                {"type", new ContentPropertyDefinition("text") }
+            };
+            var pattern = new PatternDefinition("content/file.txt");
+            var expression = new PatternExpression(pattern);
+            var path = "content/file.txt";
+            var result = expression.Match(path, propertyDefinitions);
+            Assert.NotNull(result);
+            Assert.Equal(path, result.Path);
+        }
+
+        [Fact]
+        public void LiteralSegment_OnlyShouldNotMatch()
+        {
+            // should not match the literal segment patern because file.txt != file2.txt
+            Dictionary<string, ContentPropertyDefinition> propertyDefinitions =
+                new Dictionary<string, ContentPropertyDefinition>() {
+                {"type", new ContentPropertyDefinition("text") }
+            };
+            var pattern = new PatternDefinition("content/file.txt");
+            var expression = new PatternExpression(pattern);
+            var path = "content/file2.txt";
+            var result = expression.Match(path, propertyDefinitions);
+            Assert.Null(result);
+        }
+
+        [Fact]
+        public void LiteralSegment_hasTokenSegmentShouldMatch()
+        {
+            //We have a parser for the token "name" which takes in any type of string and returns it.
+            //Therefore file.txt should be added to the result content item
+            Dictionary<string, ContentPropertyDefinition> propertyDefinitions =
+                new Dictionary<string, ContentPropertyDefinition>() {
+                {"name", new ContentPropertyDefinition("any", parser: (o, t) => o) }
+            };
+            //this pattern has both a token segment and a litteral segment
+            var pattern = new PatternDefinition("content/{name}");
+            var expression = new PatternExpression(pattern);
+            var path = "content/file.txt";
+            var match_result = expression.Match(path, propertyDefinitions);
+            object result;
+            match_result.Properties.TryGetValue("name", out result);
+            Assert.NotNull(result);
```

Comment:
Can simplify:

```suggestion
            Assert.True(match_result.Properties.TryGetValue("name", out object result));
```

This doesn't test that `result` is non-null, but the next assertion does that.